src/device/tools.jl

```diff
@@ -1,12 +1,37 @@
 # Tools for implementing device functionality
 
-# the inverse, ie. which Julia types map a given LLVM types
+# which Julia types map to a given LLVM type
+const llvmtypes = Dict{Type,Symbol}(
+    Nothing => :void,
+    Int8    => :i8,
+    Int16   => :i16,
+    Int32   => :i32,
+    Int64   => :i64,
+    Float16 => :half,
+    Float32 => :float,
+    Float64 => :double,
+)
+
+# which Julia types map to a given ROC lib type?
+const fntypes = Dict{Type,Symbol}(
+    Nothing => :void,
+    Int8    => :i8,
+    Int16   => :i16,
+    Int32   => :i32,
+    Int64   => :i64,
+    Float16 => :f16,
```

Comment:
This is sadly not true... Julia maps it to `i16`

Reply:
This is not actually intended to indicate how *Julia* maps the type, but how the suffix is mapped in the ROCm device libs.

Reply:
I'll pull this back into #6 since it's confusing being in this PR